Makefile

```diff
@@ -1,51 +1,77 @@
-OS_NAME := $(shell uname -s | tr A-Z a-z)
-EXEC = bin/LRez
-ifeq ($(OS_NAME),linux)
-        SHLIB_EXT=so
-else
-        SHLIB_EXT=dylib
+ifeq ($(SHLIB_EXT),)
+	OS_NAME := $(shell uname -s | tr A-Z a-z)
+	ifeq ($(OS_NAME),linux)
+		SHLIB_EXT=.so
+	else
+		SHLIB_EXT=.dylib
+	endif
 endif
 
 curDir = $(shell pwd)
-CC = g++
-CFLAGS  = -Wall -pedantic -O3 -m64 -shared -Wl,--whole-archive -lpthread -Wl,--no-whole-archive -std=c++11 -fPIC
 
-BAMTOOLS_INC = $(curDir)/bamtools/include/bamtools/
-BAMTOOLS_LIB = $(curDir)/bamtools/lib/
+PREFIX ?= /usr/local
+BINDIR = $(PREFIX)/bin
+LIBDIR = $(PREFIX)/lib
+
+CXX ?= g++
+CXXFLAGS += -Wall -pedantic -O3 -m64 -std=c++11 -fPIC
+LDFLAGS += -L./lib -Wl,-rpath,$(LIBDIR) -Wl,--whole-archive -lpthread -Wl,--no-whole-archive
+
+BAMTOOLS_LIB_PREFIX = lrez_
+BAMTOOLS_LIB = lib/lib$(BAMTOOLS_LIB_PREFIX)bamtools$(SHLIB_EXT)
```

Comment:
Since BamTools' build uses CMake, we can tell it to insert a custom prefix into the name, i.e., instead of building `libbamtools.so`, this produces `liblrez_bamtools.so`. This way, you can avoid collision with the upstream BamTools on the user side (e.g., we can remove the `run_constrained` in the Conda recipe with this).